GitVersionCore/BranchConfigurationCalculator.cs

```diff
@@ -0,0 +1,132 @@
+namespace GitVersion
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Text.RegularExpressions;
+    using LibGit2Sharp;
+
+    public class BranchConfigurationCalculator
```

Comment:
This is one of the worst parts of this whole approach. If anyone can come up with a better way to do it let me know.

Basically Pull Requests and Feature branches are set to Inherit the Increment. This is because they can be taken off any branch and you want them to increment the same way as the branch they were branched from.
As an example, taking a branch off master would increment patch but taking branch off develop would increment Minor.

The other option is we delete most of this code and just use the fallback which is if the repo has develop then it inherits develops config otherwise it uses master.

This code falls down if you build a feature branch which has already been merged.. That shouldn't happen, but when it does I fall back to the logic described above.

Thoughts @andreasohlund @nulltoken @SimonCropp @gep13